source/dub/recipe/packagerecipe.d

```diff
@@ -250,9 +250,16 @@ struct BuildSettingsTemplate {
 						continue;
 					}
 
-					foreach (d; dirEntries(path.toNativeString(), pattern, SpanMode.depth)) {
-						import std.path : baseName;
-						if (baseName(d.name)[0] == '.' || d.isDir) continue;
+					auto pstr = path.toNativeString();
+					foreach (d; dirEntries(pstr, pattern, SpanMode.depth)) {
+						import std.path : baseName, pathSplitter;
+						import std.algorithm.searching: canFind;
+						// eliminate any hidden files, or files in hidden directories. But always include
+						// files that are listed inside hidden directories that are specifically added to
+						// the project.
+						if (d.isDir || pathSplitter(d.name[pstr.length .. $])
+                                                                    .canFind!(name => name.length && name[0] == '.'))
```

Comment:
using spaces here but all previous lines have tabs

maybe you would like this to be like `\t\t\t\t\t\t__________.canFind` where _ is space or only tabs.

or instead of having such a long line in the first place, make a helper function to check if a given directory name should be ignored. Maybe we want to add `node_modules` to this in the future or stuff like that, having one function for it would make it easier to modify.

Reply:
I hate tabs. So my editor defaults to expanded tabs (all spaces). Fixed.

I could potentially make a separate function, or I could have used `startsWith`, which would probably be shorter. If you want me to do that, I will.